src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,431 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="HttpClientRequestMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.Serialization
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.OData.Client;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using Microsoft.OData;
+
+    /// <summary>
+    /// HttpClient based implementation of DataServiceClientRequestMessage.
+    /// </summary>
+    public class HttpClientRequestMessage : DataServiceClientRequestMessage, IDisposable

```

Comment:
Nit: Retaining the `HttpWebRequestMessage` name and namespace would have reduced the number of files changed in this PR... changing the underlying "client" should not necessarily have meant a change of the OData specific `Http*RequestMessage`

Reply:
True. But the naming could have been confusing.